homeassistant/components/weatherflow/sensor.py

```diff
@@ -46,6 +46,25 @@
 
 from .const import DOMAIN, LOGGER, format_dispatch_call
 
+DIRECTION_MAP = {
+    0: "mdi:arrow-up",  # N   (337.5° - 22.5°)
+    1: "mdi:arrow-top-right",  # NE  (22.5° - 67.5°)
+    2: "mdi:arrow-right",  # E   (67.5° - 112.5°)
+    3: "mdi:arrow-bottom-right",  # SE  (112.5° - 157.5°)
+    4: "mdi:arrow-down",  # S   (157.5° - 202.5°)
+    5: "mdi:arrow-bottom-left",  # SW  (202.5° - 247.5°)
+    6: "mdi:arrow-left",  # W   (247.5° - 292.5°)
+    7: "mdi:arrow-top-left",  # NW  (292.5° - 337.5°)
+}
+
+
+def _get_wind_direction_icon(wind_angle: float | None = None) -> str:
```

Comment:
Shouldn't the typing here be `int` if we are already doing the validation and typed `icon_fn` as such?

Reply:
This is the `icon_fn` function so it needs to return an mdi-icon hence `str`

Reply:
I meant the typing for the wind_angle input

Reply:
![image](https://github.com/user-attachments/assets/bc445645-1736-440b-9400-faff931b533c)
Ok I can get onboard with `int` for this use case.